### DISCUSSION TITLE
Becoming a Contributor

### DISCUSSION THREAD

**Discussion by raganmd** [SudoMagicCode/TouchDesigner-StyleGuide#4] [General]
If you're interested in participating this is the thread to join. From here we can connect with you about what contributing looks like and how to get started. We're currently generating this site with [Docusaurus](https://docusaurus.io/), you'll need NodeJS installed on your computer, and feel familiar with working in a git style workflow. If that's out of your comfort zone, that's okay we can walk you through what all of that looks like.

**Comment by mariamanbar** [comment]
hello hello,

I was browsing the repo and going through `docs/docs/SM-guide/050-python-in-td/`.

In my own workflow with interactive visual projects, I've noticed how quickly heavy parameter expressions can tank the network's frame rate if they aren't optimized. The current section could benefit from a few quick, clean guidelines on performance optimization, specifically around caching operator references and offloading heavy expressions to Execute DATs to avoid unnecessary cooking.

I'd love to contribute a PR with some best practices on this if you're open to it.

**Reply by raganmd** [reply]
hi @mariamanbar we're always open to having other contributors. I think our preference would be if you added a section that captured your best practices / observations. etc. That would look like creating a sub dir in `docs/docs/{your-section-here}` is that a workflow that works for what you're thinking?

**Reply by elburz** [reply]
to add to what @raganmd was mentioning, I believe the intent is more towards capturing the best practices of different folks and highlighting them as separate styles, as opposed to making one ultimate best practices that everyone is negotiating/collaborating on.
